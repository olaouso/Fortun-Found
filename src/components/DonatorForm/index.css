.submitButton {
  color: #f7f9fa;
  width: 7em;
  height: 3em;
  margin-top: 1em;
  margin-bottom: 2em;
  background: #c62a88;
  border-radius: 0.5em;
  box-shadow: 0em 1em 2em #c62a88;
  transition: transform 50ms;
  border: none;
}
.submitButton:hover {
  transform: translateY(-0.5em);
}
