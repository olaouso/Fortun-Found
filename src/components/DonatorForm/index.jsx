import React, { useState } from "react";
import { Container, Form, Col, Row } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import db from "../../firebaseConfig";
import i18next from "i18next";
import swal from "sweetalert";
import "./index.css";

export default function DonatorForm() {
  const [donatorPostData, setDonatorPostData] = useState({});
  const [currentLanguage, setCurrentLanguage] = useState(i18next.language);
  i18next.on("languageChanged", (lng) => {
    setCurrentLanguage(lng);
  });
  const handlePostChange = (e, key) => {
    setDonatorPostData({ ...donatorPostData, [key]: e.target.value });
  };
  const { t } = useTranslation();

  const addDonatorPost = (e) => {
    e.preventDefault();
    db.collection("donatorPost").add({
      title: donatorPostData.title,
      lang: currentLanguage,
      type: donatorPostData.type,
      description: donatorPostData.description,
      target: donatorPostData.target,
      profileName: donatorPostData.profileName,
      about: donatorPostData.about,
      address: donatorPostData.address,
      date: donatorPostData.date,
      image:
        "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRp5nbPgcqswxMK_MlDRX81HF17TFaUPSNRtw&usqp=CAU",
    });
    swal("", t("formsection.SubmitAlert"), "success");
  };

  return (
    <div>
      <Container>
        <Form onSubmit={addDonatorPost}>
          <Form.Group>
            <Form.Label>{t("formsection.DonationType")}</Form.Label>
            <Form.Control
              required
              as="select"
              onChange={(e) => handlePostChange(e, "type")}
            >
              <option value="service">
                {t("formsection.DonationOption1")}
              </option>
              <option value="goods">{t("formsection.DonationOption2")}</option>
              <option value="money">{t("formsection.DonationOption3")}</option>
            </Form.Control>
          </Form.Group>
          <Form.Group>
            <Form.Label>{t("formsection.PostTitle")}</Form.Label>
            <Form.Control
              required
              type="text"
              placeholder={t("formsection.PostTitle")}
              onChange={(e) => handlePostChange(e, "title")}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>{t("formsection.PostDetailsForDonator")}</Form.Label>
            <Form.Control
              required
              as="textarea"
              rows={3}
              onChange={(e) => handlePostChange(e, "description")}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>{t("formsection.TargetGroup")} </Form.Label>
            <Form.Control
              required
              as="textarea"
              rows={3}
              onChange={(e) => handlePostChange(e, "target")}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>{t("formsection.ProfileName")} </Form.Label>
            <Form.Control
              required
              type="text"
              placeholder={t("formsection.ProfileName")}
              onChange={(e) => handlePostChange(e, "profileName")}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>{t("formsection.AboutYou")}</Form.Label>
            <Form.Control
              required
              as="textarea"
              rows={3}
              onChange={(e) => handlePostChange(e, "about")}
            />
          </Form.Group>
          <Form.Group>
            <Row>
              <Col>
                <Form.Label>{t("formsection.City")}</Form.Label>
                <Form.Control
                  required
                  type="text"
                  placeholder={t("formsection.City")}
                  onChange={(e) => handlePostChange(e, "address")}
                />
              </Col>
              <Col>
                <Form.Label>{t("formsection.ApplicationDate")}</Form.Label>
                <br />
                <Form.Control
                  required
                  type="date"
                  onChange={(e) => handlePostChange(e, "date")}
                />
              </Col>
            </Row>
          </Form.Group>
          <Row className="justify-content-md-center">
            <button className="submitButton" type="submit">
              {t("formsection.Submit")}
            </button>
          </Row>
        </Form>
      </Container>
    </div>
  );
}
