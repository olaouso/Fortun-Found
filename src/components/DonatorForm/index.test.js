import React from "react";
import renderer from "react-test-renderer";
import DonatorForm from "./index.jsx";

it("renders SuccessStories correctly", () => {
  const tree = renderer.create(<DonatorForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
