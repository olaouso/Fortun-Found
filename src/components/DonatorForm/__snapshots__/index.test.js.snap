// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SuccessStories correctly 1`] = `
<div>
  <div
    className="container"
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          formsection.DonationType
        </label>
        <select
          className="form-control"
          onChange={[Function]}
          required={true}
        >
          <option
            value="service"
          >
            formsection.DonationOption1
          </option>
          <option
            value="goods"
          >
            formsection.DonationOption2
          </option>
          <option
            value="money"
          >
            formsection.DonationOption3
          </option>
        </select>
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          formsection.PostTitle
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="formsection.PostTitle"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          formsection.PostDetailsForDonator
        </label>
        <textarea
          className="form-control"
          onChange={[Function]}
          required={true}
          rows={3}
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          formsection.TargetGroup
           
        </label>
        <textarea
          className="form-control"
          onChange={[Function]}
          required={true}
          rows={3}
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          formsection.ProfileName
           
        </label>
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="formsection.ProfileName"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="form-label"
        >
          formsection.AboutYou
        </label>
        <textarea
          className="form-control"
          onChange={[Function]}
          required={true}
          rows={3}
        />
      </div>
      <div
        className="form-group"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            <label
              className="form-label"
            >
              formsection.City
            </label>
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="formsection.City"
              required={true}
              type="text"
            />
          </div>
          <div
            className="col"
          >
            <label
              className="form-label"
            >
              formsection.ApplicationDate
            </label>
            <br />
            <input
              className="form-control"
              onChange={[Function]}
              required={true}
              type="date"
            />
          </div>
        </div>
      </div>
      <div
        className="justify-content-md-center row"
      >
        <button
          className="submitButton"
          type="submit"
        >
          formsection.Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
